### ISSUE TITLE
Autocode 1000 limit now being enforced

### ISSUE THREAD

**Issue by Iangh59** [balsimpson/alexa-radio#30]
Autocode has recently started enforcing the 1000 requests limit, which makes the skill unusable as the 1000 lasts a matter of days. @balsimpson could I implore to work your magic and find a free alternative and port the code? Thanks for your work, your skill is one of my favourites.

**Comment by balsimpson** [comment]
This is a work-in-progress, but you can test it out.

https://dev.to/balsimpson/alexa-radio-skill-fle-temp-slug-7195018?preview=7aa1db5a4b9cdbd3c34086cb5356be53e8536935e458a14da35609ea5d28dc2ee9705c725542d6d60d7f4412ed0ac96f0a9b56dec9affa9031a68bed

Fully hosted on AWS, responsive website on Vercel and Firebase to store data. Any feedback/ comment is welcome. Few things might change.

**Comment by Nekosan78** [comment]
Hi Bal, thank you for replying so quickly, keen to give it a try! 

I am currently stuck on step 2, vercel site deployment. For some reason I do not have permissions to edit environment variables in the project configuration section. Or is that something that needs to be edited in the code?

![image](https://user-images.githubusercontent.com/76649055/214410141-d0543f59-0d23-4ce7-becf-616b75f1aee5.png)

**Comment by balsimpson** [comment]
You completed the previous step on the same page of Creating a GitHub repository? Once you have created it, the Environment Variables field will get activated.

![](https://res.cloudinary.com/practicaldev/image/fetch/s--or_aTG-t--/c_limit%2Cf_auto%2Cfl_progressive%2Cq_auto%2Cw_880/https://dev-to-uploads.s3.amazonaws.com/uploads/articles/ezkk2gxbnfkqb9x4gw5w.png)

**Comment by Nekosan78** [comment]
Thank you, that was the one button I did not click! I have now followed all other steps but cannot add any channels / stations to my Vercel site -  I enter all the details as per the instructions and click save, a pop up confirm it was added, but the site itself continues to say my library is empty. Any steps I may have missed?

**Comment by balsimpson** [comment]
on your Firebase Console, check if the Channels are being added in Firestore. Also check the developer console in the browser for any error messages.

**Comment by Nekosan78** [comment]
I'm afraid no channels being created in Firestore. Developer console in Vercel? No logs in the functions section.

**Comment by Iangh59** [comment]
Autocode has changed (again) how it measures usage, but I reckon with the 1000GB-s/month I have at least 300hrs of watching news channels, which is way more than I need. @Nekosan78 how many hours a month do you use the skill?
For new users, because autocode has changed something, you need to use the autocode used here, https://github.com/BitOMOn/alexa-radio, which is a fork of @balsimpson.

**Comment by balsimpson** [comment]
> I'm afraid no channels being created in Firestore. Developer console in Vercel? No logs in the functions section.

Open your Vercel page - sitename.vercel.app - on Chrome or any browser, now right click on the page and choose Inspect which will show lots of things including a Console tab. Check the console tab for any messages when you create a new Channel.

**Comment by Nekosan78** [comment]
> 

Looks like it is a permissions thing. Do you think it might have been a space in the API key copied across? (normally I do watch out for stuff like that) Can I go into the code and check?

<img width="552" alt="image" src="https://user-images.githubusercontent.com/76649055/214533148-a870d68e-3172-414f-8978-6658bebbdde4.png">

@langh59 my wife uses the skill to listen to her radio station back home. Worked absolutely fine until recently. When I checked Autocode told me I had gone beyond the limit. Can't even recreate the skill on an alt because autocode now is just browser based and it does not deploy the same way.  So keen to get this new way working!

**Comment by balsimpson** [comment]
You are right. This looks like the FIREBASE_API_KEY or the FIREBASE_PROJECT_ID is not entered correctly on Vercel. Please check if it is, as you said with extra space or something.

**Comment by Nekosan78** [comment]
Hi Bal, just checked and all keys etc. are correct. I created a second project from scratch and have the same problem. I can see on the rules side that it is denying access. Is this the potential issue:

![image](https://user-images.githubusercontent.com/76649055/214666574-07e8033c-9b87-479c-b39c-5ff8c4560970.png)

**Comment by Iangh59** [comment]
> > 
> @langh59 my wife uses the skill to listen to her radio station back home. Worked absolutely fine until recently. When I checked Autocode told me I had gone beyond the limit. Can't even recreate the skill on an alt because autocode now is just browser based and it does not deploy the same way. So keen to get this new way working!

Try https://www.amazon.com/Michael-Dworkin-Radio-Browser/dp/B09MQ7LFCT. If the station isn't listed in the database you can add the url as a custom station. It uses https://www.radio-browser.info/#!/ for the database.

The fork I linked to uses the updated Autocode that got the skill working for me. But as Autocode may change the free offering again, the more the merrier different ways of deploying balsimpson's skill are welcome.

**Comment by balsimpson** [comment]
> Hi Bal, just checked and all keys etc. are correct. I created a second project from scratch and have the same problem. I can see on the rules side that it is denying access. Is this the potential issue:
> 
> ![image](https://user-images.githubusercontent.com/76649055/214666574-07e8033c-9b87-479c-b39c-5ff8c4560970.png)

yes, i think that's what it is. change `allow read, write: if false;` to `allow read, write: if true;`

**Comment by Nekosan78** [comment]
> > > 
> > 
> > 
> > @langh59 my wife uses the skill to listen to her radio station back home. Worked absolutely fine until recently. When I checked Autocode told me I had gone beyond the limit. Can't even recreate the skill on an alt because autocode now is just browser based and it does not deploy the same way. So keen to get this new way working!
> 
> Try https://www.amazon.com/Michael-Dworkin-Radio-Browser/dp/B09MQ7LFCT. If the station isn't listed in the database you can add the url as a custom station. It uses https://www.radio-browser.info/#!/ for the database.
> 
> The fork I linked to uses the updated Autocode that got the skill working for me. But as Autocode may change the free offering again, the more the merrier different ways of deploying balsimpson's skill are welcome.

That alternative app is a great workaround, thank you so much!

Still keen on getting the new deployment method to work @balsimpson 😁 spoke to some uni buddies last week and changed the code to `allow read, write//: if false;` which also did the job. I can now add radio stations and they are saved in the database. 

The next hurdle is alexa related. Voice commands to play the station are either returning an error, or eliciting no response at all (helpful not to have consistent results). Could this be related to the endpoint that was configured on your previous skill? Thanks again for all the help by the way!

**Comment by nykul** [comment]
> > > > 
> > > 
> > > 
> > > @langh59 my wife uses the skill to listen to her radio station back home. Worked absolutely fine until recently. When I checked Autocode told me I had gone beyond the limit. Can't even recreate the skill on an alt because autocode now is just browser based and it does not deploy the same way. So keen to get this new way working!
> > 
> > 
> > Try https://www.amazon.com/Michael-Dworkin-Radio-Browser/dp/B09MQ7LFCT. If the station isn't listed in the database you can add the url as a custom station. It uses https://www.radio-browser.info/#!/ for the database.
> > The fork I linked to uses the updated Autocode that got the skill working for me. But as Autocode may change the free offering again, the more the merrier different ways of deploying balsimpson's skill are welcome.
> 
> That alternative app is a great workaround, thank you so much!
> 
> Still keen on getting the new deployment method to work @balsimpson 😁 spoke to some uni buddies last week and changed the code to `allow read, write//: if false;` which also did the job. I can now add radio stations and they are saved in the database.
> 
> The next hurdle is alexa related. Voice commands to play the station are either returning an error, or eliciting no response at all (helpful not to have consistent results). Could this be related to the endpoint that was configured on your previous skill? Thanks again for all the help by the way!

Hi Nekosan, 
I am experiencing exactly the same issues as you are.
I am no good at code but could you please "kick me" in the right direction where to find `allow read, write//: if false;`?

Thank you

**Comment by balsimpson** [comment]
@Nekosan78. In the Firebase console, go to the Rules tab. 
![firestore](https://user-images.githubusercontent.com/16057289/217768929-e9659c72-6bf5-47d1-8130-ecfd1216fceb.png)

In the Firebase console, go to the Rules tab. Change rule to read  `allow read, write: if false;`
![image](https://user-images.githubusercontent.com/76649055/214666574-07e8033c-9b87-479c-b39c-5ff8c4560970.png)

Click on Publish to save the chnages.

**Comment by nykul** [comment]
> @Nekosan78. In the Firebase console, go to the Rules tab. ![firestore](https://user-images.githubusercontent.com/16057289/217768929-e9659c72-6bf5-47d1-8130-ecfd1216fceb.png)
> 
> In the Firebase console, go to the Rules tab. Change rule to read `allow read, write: if false;` ![image](https://user-images.githubusercontent.com/76649055/214666574-07e8033c-9b87-479c-b39c-5ff8c4560970.png)
> 
> Click on Publish to save the chnages.

@balsimpson Thank you. that part is now working and I can save the channels. 
However, I am again in the same boat as @Nekosan78.
When using the voice command no reply from Alexa and no action, what am I missing please?

**Comment by Iangh59** [comment]
Have people got the new Juke Box working? I fancy having a go but want to know help is available when I get stuck.:)
